nifi-toolkit/nifi-toolkit-status-history/src/main/java/org/apache/nifi/toolkit/statushistory/StatusHistoryMain.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.toolkit.statushistory;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.nifi.controller.status.history.CsvStatusHistoryDumpFactory;
+import org.apache.nifi.controller.status.history.QuestDbStatusHistoryReader;
+import org.apache.nifi.controller.status.history.JsonStatusHistoryDumpFactory;
+import org.apache.nifi.controller.status.history.StandardStatusHistoryDumpService;
+import org.apache.nifi.controller.status.history.StatusHistoryDumpService;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class StatusHistoryMain {
```

Comment:
Due to the possibly high number of components (n*10k), when dumping component info, I would suggest to organize the results into subfolders. Either based on some attribute (first 3 character of the id for example), of put every 1000 into a new folder